package csac;

message CSACFile {
  required int32 full_block_length = 1;
  required int32 seed = 2;
  required int32 sample_rate = 3;
  required int32 quant_bits = 4;
  required float overlap = 5;
  repeated Channel channels = 6;
}

message CompressedBlock {
  repeated float samples = 1;
  required bool transient = 2;
}

message Channel {
  repeated CompressedBlock blocks = 1;
}